docs/EVALUATION.md

```diff
@@ -0,0 +1,249 @@
+
+> Remark: We would need new evaluation strategies for this as we need intervals with custom bounds on both sides.
+
+```mermaid
+gantt
+    title Pipeline run
+    dateFormat YYYY-MM-DD
+    axisFormat %Y
+    section models
+        Trigger1 :milestone, 1930, 0d
+        model 1          :a1, 1930, 10y
+        Trigger2 :milestone, :after a1
+        model 2          :a2, after a1, 5y
+        Trigger3 :milestone, :after a2
+        model 3          :a3, after a2, 5y
+        Trigger4 :milestone, :after a3
+        model 4          :a4, after a3, 25y
+        Trigger5 :milestone, :after a4
+        model 5          :a5, after a4, 25y
+        Trigger6 :milestone, :after a5
+        model 6          :a6, after a5, 5y
+        Trigger7 :milestone, :after a6
+        model 7          :a7, after a6, 5y
+    section final evaluations
+        Evaluation1 :milestone, 1940, 0d
+        Evaluation2 :milestone, 1945, 0d
+        Evaluation3 :milestone, 1950, 0d
+        Evaluation4 :milestone, 1975, 0d
+        Evaluation5 :milestone, 2000, 0d
+        Evaluation6 :milestone, 2005, 0d
+        Evaluation7 :milestone, 2010, 0d
+    section final evaluation data
+        data1 :e1, 1930, 10y
+        data2 :e2, after e1, 5y
+        data3 :e3, after e2, 5y
+        data4 :e4, after e3, 25y
+        data5 :e5, after e4, 25y
+        data6 :e6, after e5, 5y
+        data7 :e7, after e6, 5y
+    section local evaluations (for model 4)
+        Evaluation4.1 :milestone, 1950, 0d
+        Evaluation4.1 :milestone, 1954, 0d
+        Evaluation4.1 :milestone, 1958, 0d
+        Evaluation4.1 :milestone, 1962, 0d
+        Evaluation4.1 :milestone, 1966, 0d
+        Evaluation4.1 :milestone, 1970, 0d
+    section local evaluation data (for model 4): +- 5 years
+        data4.1 :e4.1, 1947, 6y
+        data4.2 :e4.2, 1951, 6y
+        data4.3 :e4.3, 1955, 6y
+        data4.4 :e4.4, 1959, 6y
+        data4.5 :e4.5, 1963, 6y
+        data4.6 :e4.6, 1967, 6y
+```
+
+### Open Question
+
+1) With the approach of evaluating a model only on future data (where the model is eventually used), it might even be fair not to use a evaluation split as the samples used for evaluation are never part of the training data of a particular model. Agreed?
+
+2) What behavior do we desire when we train multiple models on the same timestamp (e.g. if we have a lot of data at a certain point in time). One option for plotting would be to weight average the different according to how many samples of this point in time were covered by the respective model. Considering the samples that are used to evaluate the models we cannot different the evaluation data of two training runs on the same timestamps (even if their batches might contain different samples)
```

Comment:
This can occur when we don't use time triggering, right? TBH a solution does not immediately come to mind, I need to think about it.

Reply:
Yes! For sample counting policies that would occur when a dataset has unusually high sample density at one point in time. Drift policies can also produce this. For now, I don't think this is too much of a concern as we can simply plot several accuracy points at one x value. Let's just try it out first and see how often it occurs and if it's worth thinking more thoroughly.